app/services/proofing/aamva/request/verification_request.rb

```diff
@@ -140,6 +140,20 @@ def user_provided_data_map
           }
         end
 
+        def state_id_number
+          south_carolina_zero_padding
+
+          applicant.state_id_data.state_id_number
+        end
+
+        def south_carolina_zero_padding
+          return unless applicant.state_id_data.state_id_jurisdiction == 'SC'
+
+          while applicant.state_id_data.state_id_number.length < 8
+            applicant.state_id_data.state_id_number = "0#{applicant.state_id_data.state_id_number}"
+          end
```

Comment:
could also do `rjust` for a single pass:

https://ruby-doc.org/core-2.4.1/String.html#method-i-rjust

```suggestion
          applicant.state_id_data.state_id_number = applicant.state_id_data.state_id_number.rjust(8, '0')
```